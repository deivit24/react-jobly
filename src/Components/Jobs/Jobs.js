import React, { useState, useEffect } from 'react';
import API from '../../API';
import Search from '../Search/Search';
import JobCard from '../Cards/JobCard';

const Jobs = () => {
  const [jobs, setJobs] = useState([]);

  useEffect(() => {
    async function getJobs() {
      let jobs = await API.getJobs();

      setJobs(jobs);
    }
    getJobs();
  }, []);

  async function handleSearch(search) {
    let jobs = await API.getJobs(search);
    setJobs(jobs);
  }

  async function apply(idx) {
    let jobId = jobs[idx].id;
    let message = await API.applyToJob(jobId);
    setJobs((j) =>
      j.map((job) => (job.id === jobId ? { ...job, state: message } : job))
    );
  }
  let none = <p>Sorry no results</p>;
  let found = jobs.map((job, idx) => (
    <div key={idx} className="col-md-6 mb-3">
      <JobCard idx={idx} item={job} handleApply={() => apply(idx)} />
    </div>
  ));
  return (
    <section className="container mt-5">
      <h3 className="mt-5 mb-3 text-center">Search for Companies </h3>
      <Search endpoint="jobs" searchFor={handleSearch} />
      <div className="row mt-5">{jobs.length ? found : none}</div>
    </section>
  );
};

export default Jobs;
