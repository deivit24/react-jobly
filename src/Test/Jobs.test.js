import React from 'react';
import { render } from '@testing-library/react';
import Jobs from '../Components/Jobs/Jobs';

describe('Testing the Jobs component', () => {
  it('renders without crashing', () => {
    render(<Jobs />);
  });

  it('matches snapshot with no jobs', () => {
    const { asFragment } = render(<Jobs />);
    expect(asFragment()).toMatchSnapshot();
  });
});
