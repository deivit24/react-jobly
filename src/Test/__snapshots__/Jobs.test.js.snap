// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Jobs component matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <section
    class="container mt-5"
  >
    <h3
      class="mt-5 mb-3 text-center"
    >
      Search for Companies 
    </h3>
    <div
      class="Search mb-4"
    >
      <form
        class="form-inline"
      >
        <input
          class="form-control form-control-lg flex-grow-1"
          name="search"
          placeholder="Enter search term.."
          value=""
        />
        <button
          class="btn btn-lg btn-primary"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="row mt-5"
    >
      <p>
        Sorry no results
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <section
    class="container mt-5"
  >
    <h3
      class="mt-5 mb-3 text-center"
    >
      Search for Companies 
    </h3>
    <div
      class="Search mb-4"
    >
      <form
        class="form-inline"
      >
        <input
          class="form-control form-control-lg flex-grow-1"
          name="search"
          placeholder="Enter search term.."
          value=""
        />
        <button
          class="btn btn-lg btn-primary"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="row mt-5"
    >
      <p>
        Sorry no results
      </p>
    </div>
  </section>
</DocumentFragment>
`;
